apps/condo/domains/notification/schema/_internalSendNotificationNewMobileAppVersionService.js

```diff
@@ -0,0 +1,151 @@
+        },
+        {
+            access: true,
+            type: 'type _internalSendNotificationNewMobileAppVersionOutput { messageBatchId: ID! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.can_internalSendNotificationNewMobileAppVersion,
+            schemaDoc: 'The mutation is necessary to send push notifications directly to a specific user device with an outdated version of the mobile application build.',
+            schema: '_internalSendNotificationNewMobileAppVersion(data: _internalSendNotificationNewMobileAppVersionInput!): _internalSendNotificationNewMobileAppVersionOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+
+                const { data: { dv, sender, platform, app, buildVersion, title, body, organizationIds } } = args
+
+                const { keystone } = await getSchemaCtx('MessageBatch')
+                const adminContext = await keystone.createContext({ skipAccessControl: true })
+
+                const userType = app === RESIDENT ? RESIDENT : STAFF
+                let remoteClientKeys
+
+                //The case when a push is sent to all suitable devices, regardless of organization
+                if (isEmpty(organizationIds)) {
+                    remoteClientKeys = await loadListByChunks({
+                        context: adminContext,
+                        list: RemoteClient,
+                        where: {
+                            devicePlatform: platform,
+                            deletedAt: null,
+                            owner: {
+                                type: userType,
+                            },
+                        },
+                        chunkSize: 50,
+                        chunkProcessor: (/** @type {RemoteClient[]} */ chunk) => {
+                            const rcWithLowerBuild = []
+
+                            if (buildVersion) {
+                                for (const rc of chunk) {
+                                    if (get(rc, 'meta.Build', buildVersion) < buildVersion) rcWithLowerBuild.push(`rc:${rc.id}`)
+                                }
+                            }
+
+                            return rcWithLowerBuild
+                        },
+                    })
+                } else {    //The case when a push is sent to suitable devices of a specific or several organizations
+                    const userIds = await loadListByChunks({
+                        context: adminContext,
+                        list: Resident,
+                        where: {
+                            organization: {
+                                id_in: organizationIds,
+                            },
+                        },
+                        chunkSize: 50,
+                        chunkProcessor: (/** @type {Resident[]} */ chunk) => {
+                            const userIds = []
+                            chunk.map(resident => {
```

Comment:
`map` should return value. You're not, so you creating redundant unused array of `undefined`

Reply:
So please use `for`-lop here (forEach is acceptable, but not recommended, since it slower)